superset/utils/core.py

```diff
@@ -1306,7 +1305,7 @@ def get_metric_name(metric: Metric) -> str:
             if column_name:
                 return column_name
         raise ValueError(__("Invalid metric object"))
-    return cast(str, metric)
+    return metric  # type: ignore
```

Comment:
For some reason `mypy` didn't pick up that `Union[str, AdhocMetric]` becomes `str` after we do `is_adhoc_metric(metric)` above